server/testkit/tests/identity_verification_tests.rs

```diff
@@ -0,0 +1,448 @@
+use core::result::Result::Err;
+use kanidm_client::KanidmClient;
+use kanidm_proto::{
+    internal::{IdentifyUserRequest, IdentifyUserResponse},
+    v1::{Entry, Filter},
+};
+use kanidmd_lib::{
+    credential::totp::{Totp, TotpAlgo, TotpDigits},
+    f_and, filter,
+    prelude::{f_eq, f_sub, m_purge},
+    value::PartialValue,
+};
+use kanidmd_testkit::ADMIN_TEST_PASSWORD;
+use openssl::{
+    ec::EcKey,
+    pkey::{PKey, Private, Public},
+    pkey_ctx::PkeyCtx,
+};
+use reqwest::{Error, StatusCode};
+use std::time::SystemTime;
+
+static UNIVERSAL_PW: &'static str = "eicieY7ahchaoCh0eeTa";
+static UNIVERSAL_PW_HASH: &'static str =
+    "pbkdf2_sha256$36000$xIEozuZVAoYm$uW1b35DUKyhvQAf1mBqMvoBDcqSD06juzyO/nmyV0+w=";
+
+static INVALID_USER_UUID: &'static str = "d2b496bd-8493-47b7-8142-f568b5cf47e0";
+static INVALID_USER_NAME: &'static str = "invalid_user";
+
+static VALID_USER_A_UUID: &'static str = "d2b496bd-8493-47b7-8142-f568b5cf47e1";
+static VALID_USER_A_NAME: &'static str = "valid_user_A";
+
+static VALID_USER_B_UUID: &'static str = "d2b496bd-8493-47b7-8142-f568b5cf47e2";
+static VALID_USER_B_NAME: &'static str = "valid_user_B";
+
+fn get_identity_verification_endpoint(other_id: &str) -> String {
+    format!("/v1/person/{other_id}/_identify_user")
+}
+
+fn get_url(addr: &str, id: &str) -> String {
+    format!("{addr}/{}", get_identity_verification_endpoint(id))
+}
+
+// UNAUTHENTICATED CASES:
+#[kanidmd_testkit::test]
+async fn test_not_authenticated(rsclient: KanidmClient) {
+    // basically here we try a bit of all the possible combinations while unauthenticated to check it's not working
+    setup_server(&rsclient).await;
+    create_valid_user_a(&rsclient).await;
+    rsclient.logout();
+    let res: Result<IdentifyUserResponse, Error> = make_request(
+        &rsclient,
+        IdentifyUserRequest::Start,
+        Some(VALID_USER_A_NAME.to_string()),
+    )
+    .await;
+    assert!(res.is_err());
+    if let Err(err) = res {
+        assert_eq!(err.status().unwrap(), StatusCode::FORBIDDEN)
+    }
```

Comment:
Spaces between the different  asserts, you can also do a single assert with:
`assert!(matches!(Err(StatusCode::FORBIDDEN), res));`

Reply:
I like the single patter more tbh